src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java

```diff
@@ -525,7 +520,7 @@ public NamespaceMapper getNamespaceMapper() {
               }
             });
 
-    if (!containsError.get()) {
+    if (results.values().stream().allMatch(v -> v.errorMessages().isEmpty())) {
```

Comment:
As a for loop over entrysets is ugly and harder to read than a forEach loop - I opted to do a value check at the end.  Given validation is a startup process there is little overhead for this.